3rdParty/velocypack/include/velocypack/velocypack-common.h

```diff
@@ -206,21 +321,91 @@ static inline int64_t toInt64(uint64_t v) noexcept {
                      : static_cast<int64_t>(v);
 }
 
+
+#ifdef _WIN32
+#pragma warning(push)
+#pragma warning(disable: 4554)
+#endif
+template <typename T, unsigned Bytes, unsigned Shift = 0>
+static inline T readIntFixed(uint8_t const* p) noexcept {
+  // bailout if nothing to shift or target type is too small for shift
+  // to avoid compiler warning
+  if constexpr (Bytes == 0 || sizeof(T) * 8 <= Shift) {
+      return 0;
+  } else {
+    return readIntFixed<T, Bytes - 1, Shift + 8>(p + 1) | (static_cast<T>(*p) << Shift);
+    // for some reason MSVC detects possible operator precedence error here ~~^                                                                          ^
+  }
+}
+#ifdef _WIN32
+#pragma warning(pop)
+#endif
+
+template <typename T>
+static inline T readIntegerFixed(uint8_t const* start, uint8_t length) noexcept {
+  switch (length) {
+    case 1:
+      return *start;
+    case 2:
+      return readIntFixed<T, 2>(start);
+    case 3:
+      return readIntFixed<T, 3>(start);
+    case 4:
+      return readIntFixed<T, 4>(start);
+    case 5:
+      return readIntFixed<T, 5>(start);
+    case 6:
+      return readIntFixed<T, 6>(start);
+    case 7:
+      return readIntFixed<T, 7>(start);
+    case 8: {
+      // avoid compiler warning for small types
+      if constexpr (sizeof(T) == 8) {
+        T v;
+        memcpy(&v, start, 8);
+        return v;
```

Comment:
We have to swap bytes for BE platforms

Reply:
Indeed. Forgot this place. But in  the next commit this will be rewritten as we agreed.